py/desitarget/mtl.py

```diff
@@ -70,13 +75,12 @@ def make_mtl(targets, obscon, zcat=None, trim=False, scnd=None):
         # APC Propagate a flag on which targets came from scnd
         is_scnd = np.repeat(False, len(targets))
         is_scnd[-nrows:] = True
+        # APC Assign the SCND_ANY bit to secondary targets. We need to do this
+        # APC before the NUMOBS_MORE logic below
+        log.info('Updating DESI_TARGET for secondaries...')
+        targets[desi_target][is_scnd] |= desi_mask.SCND_ANY
```

Comment:
I'm nervous about updating target bits within MTL itself. The target bits should be updated in the `cuts/secondary` functions, and MTL should just respond to those bits.

Could you assign the `SCND_ANY` bit to the relevant targets in `secondary.py` instead of in `mtl.py`? This would mean including the `DESI_TARGET` output column for the secondary files, but I'm OK with that. 

Sorry! It's just cleaner to only update the target bits outside of MTL.

Reply:
OK -- sorry I might not have time to get this done and tested today. 

For my own reference (@geordie666 , would be useful to have a second opinion): on quick inspection it seems like I need to add a `DESI_TARGET` column to the `outdatamodel` in `secondary.py` and then populate this with `SCND_ANY` somewhere, maybe in the loop over input files in `read_files`.

I need to make sure the `DESI_TARGET` column doesn't overwrite anything for  the primaries and ends up with the right 'SV_' or whatever prefix. 

In the `|` vs `&` changes elsewhere in this PR I'm using `from desitarget.targetmask import desi_mask` to get the mask that will give `SCND_ANY`. Probably I shouldn't be doing this but rather getting the program-specific `desi_mask`?

Reply:
Slightly more defensible than adding the `DESI_TARGET` column in MTL might be to do it when we write the files? That's not ideal, but we already add a lot of necessary information on the write (such as the `SUBPRIORITY` column) so I think we're stuck with that model.

This line:

https://github.com/desihub/desitarget/blob/f6369d5bcd04f536e0087068e971d410fc91a513/py/desitarget/io.py#L725

already gives you the correct output column (just change the `_` to `cols` and then use `cols[0]` as `DESI_TARGET` (or, it should correctly also revert to `SV1_DESI_TARGET` when appropriate).

Best, I think, would be to add the `DESI_TARGET` column in `secondary.finalize_secondary()`. Maybe just above this line:

https://github.com/desihub/desitarget/blob/f6369d5bcd04f536e0087068e971d410fc91a513/py/desitarget/secondary.py#L731-L732

I _think_ my `main_cmx_or_sv` trick from:

https://github.com/desihub/desitarget/blob/f6369d5bcd04f536e0087068e971d410fc91a513/py/desitarget/io.py#L725

will work here, too (although it depends on which exact columns are set).

My preference would be to add `DESI_TARGET` in `secondary.finalize_secondary()`, but if that proves convoluted, I'd be OK with adding it in `io.write_secondary()`, too

Reply:
Much better to add `SCND_ANY` and `DESI_TARGET` in the new locations. Thanks.